File: JavaScript/프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.js
function solution(s) {
    var answer = [];
    const pmap = new Map();
    for (let i = 0; i < s.length; i++) {
        if (pmap.has(s[i])) {
            answer.push(i - pmap.get(s[i]))
        }
        else {
            answer.push(-1)
        }
        pmap.set(s[i],i)
    }
    return answer;
}

File: Java/프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.java
import java.util.*;

class Solution {
    public int[] solution(String s) {
        int[] answer = new int[s.length()];
        Map<Character, Integer> pmap = new HashMap<>();
        for (int i=0; i < s.length(); i++) {
            if (pmap.containsKey(s.charAt(i))) {
                answer[i] = i - pmap.get(s.charAt(i));
            }
            else {
                answer[i] = -1;
            }
            pmap.put(s.charAt(i), i);
        }
        return answer;
    }
}